caddyhttp/httpserver/replacer.go

```diff
@@ -223,8 +223,22 @@ func (r *replacer) getSubstitution(key string) string {
 		}
 		return host
 	case "{path}":
-		return r.request.URL.Path
+		// if a rewrite has happened, the original URI should be used as the path
+		// rather than the rewritten URI
+		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
+		if path == "" {
+			path = r.request.URL.Path
+		}
+		return path
 	case "{path_escaped}":
+		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
```

Comment:
For a value that is shared between middlewares, probably should be in the `httpserver` package, which is this one. I am still wondering if context would be a better way to do this...